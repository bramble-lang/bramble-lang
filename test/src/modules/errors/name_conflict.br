
fn my_main() -> i64 {
    let ms: MyStruct := my_mod::g();
    return 0;
}

struct MyStruct {
    b: i64,
}

fn f() -> MyStruct {
    return MyStruct{b: 6};
}

mod my_mod {
    struct MyStruct {
        a: i64,
    }

    fn g() -> MyStruct {
        return MyStruct{
            a: 5,
        };
    }
}