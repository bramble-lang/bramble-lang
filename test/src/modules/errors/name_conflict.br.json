[{"id": 1, "stage": "lexer", "source": [1, 3], "ok": "Keyword"},
{"id": 2, "stage": "lexer", "source": [4, 11], "ok": "Identifier"},
{"id": 3, "stage": "lexer", "source": [11, 12], "ok": "Operator"},
{"id": 4, "stage": "lexer", "source": [12, 13], "ok": "Operator"},
{"id": 5, "stage": "lexer", "source": [14, 16], "ok": "Operator"},
{"id": 6, "stage": "lexer", "source": [17, 20], "ok": "Primitive"},
{"id": 7, "stage": "lexer", "source": [21, 22], "ok": "Operator"},
{"id": 8, "stage": "lexer", "source": [27, 30], "ok": "Keyword"},
{"id": 9, "stage": "lexer", "source": [31, 33], "ok": "Identifier"},
{"id": 10, "stage": "lexer", "source": [33, 34], "ok": "Operator"},
{"id": 11, "stage": "lexer", "source": [35, 43], "ok": "Identifier"},
{"id": 12, "stage": "lexer", "source": [44, 46], "ok": "Operator"},
{"id": 13, "stage": "lexer", "source": [47, 53], "ok": "Identifier"},
{"id": 14, "stage": "lexer", "source": [53, 55], "ok": "Operator"},
{"id": 15, "stage": "lexer", "source": [55, 56], "ok": "Identifier"},
{"id": 16, "stage": "lexer", "source": [56, 57], "ok": "Operator"},
{"id": 17, "stage": "lexer", "source": [57, 58], "ok": "Operator"},
{"id": 18, "stage": "lexer", "source": [58, 59], "ok": "Operator"},
{"id": 19, "stage": "lexer", "source": [64, 70], "ok": "Keyword"},
{"id": 20, "stage": "lexer", "source": [71, 72], "ok": "Integer"},
{"id": 21, "stage": "lexer", "source": [72, 73], "ok": "Operator"},
{"id": 22, "stage": "lexer", "source": [74, 75], "ok": "Operator"},
{"id": 23, "stage": "lexer", "source": [77, 83], "ok": "Keyword"},
{"id": 24, "stage": "lexer", "source": [84, 92], "ok": "Identifier"},
{"id": 25, "stage": "lexer", "source": [93, 94], "ok": "Operator"},
{"id": 26, "stage": "lexer", "source": [99, 100], "ok": "Identifier"},
{"id": 27, "stage": "lexer", "source": [100, 101], "ok": "Operator"},
{"id": 28, "stage": "lexer", "source": [102, 105], "ok": "Primitive"},
{"id": 29, "stage": "lexer", "source": [105, 106], "ok": "Operator"},
{"id": 30, "stage": "lexer", "source": [107, 108], "ok": "Operator"},
{"id": 31, "stage": "lexer", "source": [110, 112], "ok": "Keyword"},
{"id": 32, "stage": "lexer", "source": [113, 114], "ok": "Identifier"},
{"id": 33, "stage": "lexer", "source": [114, 115], "ok": "Operator"},
{"id": 34, "stage": "lexer", "source": [115, 116], "ok": "Operator"},
{"id": 35, "stage": "lexer", "source": [117, 119], "ok": "Operator"},
{"id": 36, "stage": "lexer", "source": [120, 128], "ok": "Identifier"},
{"id": 37, "stage": "lexer", "source": [129, 130], "ok": "Operator"},
{"id": 38, "stage": "lexer", "source": [135, 141], "ok": "Keyword"},
{"id": 39, "stage": "lexer", "source": [142, 150], "ok": "Identifier"},
{"id": 40, "stage": "lexer", "source": [150, 151], "ok": "Operator"},
{"id": 41, "stage": "lexer", "source": [151, 152], "ok": "Identifier"},
{"id": 42, "stage": "lexer", "source": [152, 153], "ok": "Operator"},
{"id": 43, "stage": "lexer", "source": [154, 155], "ok": "Integer"},
{"id": 44, "stage": "lexer", "source": [155, 156], "ok": "Operator"},
{"id": 45, "stage": "lexer", "source": [156, 157], "ok": "Operator"},
{"id": 46, "stage": "lexer", "source": [158, 159], "ok": "Operator"},
{"id": 47, "stage": "lexer", "source": [161, 164], "ok": "Keyword"},
{"id": 48, "stage": "lexer", "source": [165, 171], "ok": "Identifier"},
{"id": 49, "stage": "lexer", "source": [172, 173], "ok": "Operator"},
{"id": 50, "stage": "lexer", "source": [178, 184], "ok": "Keyword"},
{"id": 51, "stage": "lexer", "source": [185, 193], "ok": "Identifier"},
{"id": 52, "stage": "lexer", "source": [194, 195], "ok": "Operator"},
{"id": 53, "stage": "lexer", "source": [204, 205], "ok": "Identifier"},
{"id": 54, "stage": "lexer", "source": [205, 206], "ok": "Operator"},
{"id": 55, "stage": "lexer", "source": [207, 210], "ok": "Primitive"},
{"id": 56, "stage": "lexer", "source": [210, 211], "ok": "Operator"},
{"id": 57, "stage": "lexer", "source": [216, 217], "ok": "Operator"},
{"id": 58, "stage": "lexer", "source": [223, 225], "ok": "Keyword"},
{"id": 59, "stage": "lexer", "source": [226, 227], "ok": "Identifier"},
{"id": 60, "stage": "lexer", "source": [227, 228], "ok": "Operator"},
{"id": 61, "stage": "lexer", "source": [228, 229], "ok": "Operator"},
{"id": 62, "stage": "lexer", "source": [230, 232], "ok": "Operator"},
{"id": 63, "stage": "lexer", "source": [233, 241], "ok": "Identifier"},
{"id": 64, "stage": "lexer", "source": [242, 243], "ok": "Operator"},
{"id": 65, "stage": "lexer", "source": [252, 258], "ok": "Keyword"},
{"id": 66, "stage": "lexer", "source": [259, 267], "ok": "Identifier"},
{"id": 67, "stage": "lexer", "source": [267, 268], "ok": "Operator"},
{"id": 68, "stage": "lexer", "source": [281, 282], "ok": "Identifier"},
{"id": 69, "stage": "lexer", "source": [282, 283], "ok": "Operator"},
{"id": 70, "stage": "lexer", "source": [284, 285], "ok": "Integer"},
{"id": 71, "stage": "lexer", "source": [285, 286], "ok": "Operator"},
{"id": 72, "stage": "lexer", "source": [295, 296], "ok": "Operator"},
{"id": 73, "stage": "lexer", "source": [296, 297], "ok": "Operator"},
{"id": 74, "stage": "lexer", "source": [302, 303], "ok": "Operator"},
{"id": 75, "stage": "lexer", "source": [304, 305], "ok": "Operator"},
{"id": 81, "parent_id": 79, "stage": "parser", "source": [17, 20], "ok": "Primitive Type"},
{"id": 79, "parent_id": 78, "stage": "parser", "source": [1, 20], "ok": "Routine Declaration"},
{"id": 86, "parent_id": 85, "stage": "parser", "source": [35, 43], "ok": "Path"},
{"id": 85, "parent_id": 84, "stage": "parser", "source": [35, 43], "ok": "Custom Type"},
{"id": 84, "parent_id": 83, "stage": "parser", "source": [31, 43], "ok": "Identifier Declaration"},
{"id": 97, "parent_id": 96, "stage": "parser", "source": [47, 56], "ok": "Path"},
{"id": 96, "parent_id": 94, "stage": "parser", "source": [47, 58], "ok": "Routine Call"},
{"id": 94, "parent_id": 92, "stage": "parser", "source": [47, 58]},
{"id": 92, "parent_id": 91, "stage": "parser", "source": [47, 58]},
{"id": 91, "parent_id": 90, "stage": "parser", "source": [47, 58]},
{"id": 90, "parent_id": 89, "stage": "parser", "source": [47, 58]},
{"id": 89, "parent_id": 88, "stage": "parser", "source": [47, 58]},
{"id": 88, "parent_id": 83, "stage": "parser", "source": [47, 58]},
{"id": 83, "parent_id": 82, "stage": "parser", "source": [27, 58], "ok": "Let Binding"},
{"id": 82, "parent_id": 78, "stage": "parser", "source": [27, 59], "ok": "Statement Bind"},
{"id": 138, "parent_id": 134, "stage": "parser", "source": [71, 72], "ok": "Number"},
{"id": 134, "parent_id": 132, "stage": "parser", "source": [71, 72]},
{"id": 132, "parent_id": 131, "stage": "parser", "source": [71, 72]},
{"id": 131, "parent_id": 130, "stage": "parser", "source": [71, 72]},
{"id": 130, "parent_id": 129, "stage": "parser", "source": [71, 72]},
{"id": 129, "parent_id": 128, "stage": "parser", "source": [71, 72]},
{"id": 128, "parent_id": 127, "stage": "parser", "source": [71, 72]},
{"id": 127, "parent_id": 78, "stage": "parser", "source": [64, 72], "ok": "Return"},
{"id": 78, "parent_id": 76, "stage": "parser", "source": [1, 75], "ok": "Funtion Definition"},
{"id": 141, "parent_id": 140, "stage": "parser", "source": [102, 105], "ok": "Primitive Type"},
{"id": 140, "parent_id": 139, "stage": "parser", "source": [99, 105], "ok": "Identifier Declaration"},
{"id": 139, "parent_id": 76, "stage": "parser", "source": [77, 108], "ok": "Struct Definition"},
{"id": 149, "parent_id": 148, "stage": "parser", "source": [120, 128], "ok": "Path"},
{"id": 148, "parent_id": 146, "stage": "parser", "source": [120, 128], "ok": "Custom Type"},
{"id": 146, "parent_id": 145, "stage": "parser", "source": [110, 128], "ok": "Routine Declaration"},
{"id": 176, "parent_id": 175, "stage": "parser", "source": [142, 150], "ok": "Path"},
{"id": 188, "parent_id": 184, "stage": "parser", "source": [154, 155], "ok": "Number"},
{"id": 184, "parent_id": 182, "stage": "parser", "source": [154, 155]},
{"id": 182, "parent_id": 181, "stage": "parser", "source": [154, 155]},
{"id": 181, "parent_id": 180, "stage": "parser", "source": [154, 155]},
{"id": 180, "parent_id": 179, "stage": "parser", "source": [154, 155]},
{"id": 179, "parent_id": 178, "stage": "parser", "source": [154, 155]},
{"id": 178, "parent_id": 177, "stage": "parser", "source": [154, 155]},
{"id": 177, "parent_id": 175, "stage": "parser", "source": [150, 156], "ok": "Struct Exprssion Parameters"},
{"id": 175, "parent_id": 173, "stage": "parser", "source": [142, 156], "ok": "Struct Expression"},
{"id": 173, "parent_id": 171, "stage": "parser", "source": [142, 156]},
{"id": 171, "parent_id": 170, "stage": "parser", "source": [142, 156]},
{"id": 170, "parent_id": 169, "stage": "parser", "source": [142, 156]},
{"id": 169, "parent_id": 168, "stage": "parser", "source": [142, 156]},
{"id": 168, "parent_id": 167, "stage": "parser", "source": [142, 156]},
{"id": 167, "parent_id": 166, "stage": "parser", "source": [142, 156]},
{"id": 166, "parent_id": 145, "stage": "parser", "source": [135, 156], "ok": "Return"},
{"id": 145, "parent_id": 76, "stage": "parser", "source": [110, 159], "ok": "Funtion Definition"},
{"id": 197, "parent_id": 196, "stage": "parser", "source": [207, 210], "ok": "Primitive Type"},
{"id": 196, "parent_id": 195, "stage": "parser", "source": [204, 210], "ok": "Identifier Declaration"},
{"id": 195, "parent_id": 191, "stage": "parser", "source": [178, 217], "ok": "Struct Definition"},
{"id": 205, "parent_id": 204, "stage": "parser", "source": [233, 241], "ok": "Path"},
{"id": 204, "parent_id": 202, "stage": "parser", "source": [233, 241], "ok": "Custom Type"},
{"id": 202, "parent_id": 201, "stage": "parser", "source": [223, 241], "ok": "Routine Declaration"},
{"id": 232, "parent_id": 231, "stage": "parser", "source": [259, 267], "ok": "Path"},
{"id": 244, "parent_id": 240, "stage": "parser", "source": [284, 285], "ok": "Number"},
{"id": 240, "parent_id": 238, "stage": "parser", "source": [284, 285]},
{"id": 238, "parent_id": 237, "stage": "parser", "source": [284, 285]},
{"id": 237, "parent_id": 236, "stage": "parser", "source": [284, 285]},
{"id": 236, "parent_id": 235, "stage": "parser", "source": [284, 285]},
{"id": 235, "parent_id": 234, "stage": "parser", "source": [284, 285]},
{"id": 234, "parent_id": 233, "stage": "parser", "source": [284, 285]},
{"id": 233, "parent_id": 231, "stage": "parser", "source": [267, 296], "ok": "Struct Exprssion Parameters"},
{"id": 231, "parent_id": 229, "stage": "parser", "source": [259, 296], "ok": "Struct Expression"},
{"id": 229, "parent_id": 227, "stage": "parser", "source": [259, 296]},
{"id": 227, "parent_id": 226, "stage": "parser", "source": [259, 296]},
{"id": 226, "parent_id": 225, "stage": "parser", "source": [259, 296]},
{"id": 225, "parent_id": 224, "stage": "parser", "source": [259, 296]},
{"id": 224, "parent_id": 223, "stage": "parser", "source": [259, 296]},
{"id": 223, "parent_id": 222, "stage": "parser", "source": [259, 296]},
{"id": 222, "parent_id": 201, "stage": "parser", "source": [252, 296], "ok": "Return"},
{"id": 201, "parent_id": 191, "stage": "parser", "source": [223, 303], "ok": "Funtion Definition"},
{"id": 191, "parent_id": 76, "stage": "parser", "source": [161, 305], "ok": "Module"},
{"id": 76, "stage": "parser", "source": [1, 305], "ok": "File Module"},
{"id": 256, "stage": "canonize-item-path", "source": [0, 305], "ok": "$name_conflict::name_conflict"},
{"id": 257, "stage": "canonize-item-path", "source": [1, 305], "ok": "$name_conflict::name_conflict::name_conflict"},
{"id": 258, "stage": "canonize-item-path", "source": [161, 305], "ok": "$name_conflict::name_conflict::my_mod::my_mod"},
{"id": 259, "stage": "canonize-item-path", "source": [223, 303], "ok": "$name_conflict::name_conflict::my_mod::g"},
{"id": 260, "stage": "canonize-item-path", "source": [178, 217], "ok": "$name_conflict::name_conflict::my_mod::MyStruct"},
{"id": 261, "stage": "canonize-item-path", "source": [204, 210], "ok": "$name_conflict::name_conflict::my_mod::a"},
{"id": 262, "stage": "canonize-item-path", "source": [1, 75], "ok": "$name_conflict::name_conflict::my_main"},
{"id": 263, "stage": "canonize-item-path", "source": [110, 159], "ok": "$name_conflict::name_conflict::f"},
{"id": 264, "stage": "canonize-item-path", "source": [77, 108], "ok": "$name_conflict::name_conflict::MyStruct"},
{"id": 265, "stage": "canonize-item-path", "source": [99, 105], "ok": "$name_conflict::name_conflict::b"},
{"id": 266, "stage": "canonize-type-ref", "source": [223, 303], "ok": "$name_conflict::name_conflict::my_mod::MyStruct"},
{"id": 267, "stage": "canonize-type-ref", "source": [259, 296], "ok": "$name_conflict::name_conflict::my_mod::MyStruct"},
{"id": 268, "stage": "canonize-type-ref", "source": [27, 59], "ok": "$name_conflict::name_conflict::MyStruct"},
{"id": 269, "stage": "canonize-type-ref", "source": [47, 58], "ok": "$name_conflict::name_conflict::my_mod::g"},
{"id": 270, "stage": "canonize-type-ref", "source": [110, 159], "ok": "$name_conflict::name_conflict::MyStruct"},
{"id": 271, "stage": "canonize-type-ref", "source": [142, 156], "ok": "$name_conflict::name_conflict::MyStruct"},
{"id": 275, "parent_id": 274, "stage": "type-resolver", "source": [284, 285], "ok": "i64", "addressability": "None"},
{"id": 274, "parent_id": 273, "stage": "type-resolver", "source": [259, 296], "ok": "$name_conflict::name_conflict::my_mod::MyStruct", "addressability": "None", "ref": [178, 217]},
{"id": 273, "parent_id": 272, "stage": "type-resolver", "source": [252, 296], "ok": "$name_conflict::name_conflict::my_mod::MyStruct", "addressability": "None"},
{"id": 272, "stage": "type-resolver", "source": [223, 303], "ok": "$name_conflict::name_conflict::my_mod::MyStruct", "addressability": "None"},
{"id": 277, "stage": "type-resolver", "source": [204, 210], "ok": "i64", "addressability": "None"},
{"id": 276, "stage": "type-resolver", "source": [178, 217], "ok": "Unit", "addressability": "None"},
{"id": 280, "parent_id": 279, "stage": "type-resolver", "source": [47, 58], "ok": "$name_conflict::name_conflict::my_mod::MyStruct", "addressability": "None", "ref": [223, 303]},
{"id": 279, "parent_id": 278, "stage": "type-resolver", "source": [27, 59], "error": "L3: Bind expected $name_conflict::name_conflict::MyStruct but got $name_conflict::name_conflict::my_mod::MyStruct"},
{"id": 278, "stage": "type-resolver", "source": [27, 59], "error": "L3: Bind expected $name_conflict::name_conflict::MyStruct but got $name_conflict::name_conflict::my_mod::MyStruct"}]