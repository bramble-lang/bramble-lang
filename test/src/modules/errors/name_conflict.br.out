Error: L3: Bind expected $name_conflict::name_conflict::MyStruct but got $name_conflict::name_conflict::my_mod::MyStruct
